include/core/physical_property_model.h

```diff
@@ -36,15 +36,29 @@ enum field : int
   previous_temperature
 };
 
+inline void
+set_field_vector(const field                          id,
+                 const std::vector<double> &          data,
+                 std::map<field, std::vector<double>> fields)
+{
+  std::vector<double> &target = fields.at(id);
```

Comment:
Does this mean that the given fields parameter already contains an element for each possible id (i.e. shear_rate, temperature and previous_temperature)? What I mean is, when this function will be called, is the fields parameters something that looks like this:
`{
{shear_rate, []},
{temperature, []},
{previous_temperature, []},
}` ?
If yes, why would it already contain all 3 fields? If not, the at() method will not work if the field is not already existing. The [ ] accessor on the other had will get the proper field if already existing, but will create it if it does not exist. We'll discuss that tomorrow :)

Reply:
I think I better understand now that the specified id has to be in the fields.

Reply:
It's a very good point. The field vector has to have that memory pre-allocated, otherwise it won't work. I faced a challenge here. I wanted to have a flexible structure, but then I ended up not wanting to do dynamic memory allocation in the scratch reinit, because that's the whole point of the scratch. So my best solution was to allocate the memory from the get_go for the scratch, but only fill the fields that we would need.